File: models/character.py
import uuid

class Character:
    def __init__(self, name, classType="Explorer", lvl=1):
        self.characterId = uuid.uuid4()
        self.name = name
        self.classType = classType
        self.lvl = lvl
        self.inventory = []  

    def levelUp(self):
        self.lvl += 1

    def addItem(self, item):
        self.inventory.append(item)

    def removeItem(self, item):
        if item in self.inventory:
            self.inventory.remove(item)

    def __str__(self):
        return f"Character(name={self.name}, class={self.classType}, level={self.level})"

File: INF122Assignment3/src/Character.java
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
public class Character {
    private UUID character_id;
    private String name;
    private String class_type;
    private int level;
    private List<InventoryItem> inventory;

    public Character(String name, String class_type){
        this.character_id = UUID.randomUUID();
        this.name = name;
        this.class_type = class_type;
        this.level = 0;
        this.inventory = new ArrayList<>();
        inventory.add(new InventoryItem("Wooden Sword", "A basic wooden sword. Does 5 damage per hit.", "common", 1));
    }

    public void levelUp(){
        this.level++;
    }

    public void addItem(InventoryItem item){
        this.inventory.add(item);
    }

    public InventoryItem findItem(String name){
        for (InventoryItem i : this.inventory){
            if(i.getName().equalsIgnoreCase(name)){
                return i;
            }
        }
        return null;
    }

    public boolean removeItem(String name){
        InventoryItem item = findItem(name);
        if(item != null){
            if(item.getQuantity() > 1){
                item.setQuantity(item.getQuantity() - 1);
            }
            else this.inventory.remove(item);
            return true;
        }
        return false;
    }

    public String toString(){
        return String.format("%s, %s\n Level: %s", this.name, this.class_type, this.level);
    }

    public void displayInventory(){
        if(this.inventory.isEmpty()) System.out.println("Your inventory is empty!");
        for(InventoryItem item : this.inventory){
            System.out.println(item);
        }
    }
}
